src/Attachment.js

```diff
@@ -0,0 +1,176 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+/**
+TinCan client library
+
+@module TinCan
+@submodule TinCan.Attachment
+**/
+(function () {
+    "use strict";
+
+    /**
+    @class TinCan.Attachment
+    @constructor
+    */
+    var Attachment = TinCan.Attachment = function (cfg) {
+        this.log("constructor");
+
+        /**
+        @property usageType
+        @type String
+        */
+        this.usageType = null;
+
+        /**
+        @property display
+        @type Object
+        */
+        this.display = null;
+
+        /**
+        @property contentType
+        @type String
+        */
+        this.contentType = null;
+
+        /**
+        @property length
+        @type int
+        */
+        this.length = null;
+
+        /**
+        @property sha2
+        @type String
+        */
+        this.sha2 = null;
+
+        /**
+        @property description
+        @type Object
+        */
+        this.description = null;
+
+        /**
+        @property fileUrl
+        @type String
+        */
+        this.fileUrl = null;
+
+        this.init(cfg);
+    };
+    Attachment.prototype = {
+        /**
+        @property LOG_SRC
+        */
+        LOG_SRC: "Attachment",
+
+        /**
+        @method log
+        */
+        log: TinCan.prototype.log,
+
+        /**
+        @method init
+        @param {Object} [options] Configuration used to initialize
+        */
+        init: function (cfg) {
+            this.log("init");
+            var i,
+                directProps = [
+                    "contentType",
+                    "length",
+                    "sha2",
+                    "usageType",
+                    "display"
+                ]
+            ;
+
+            cfg = cfg || {};
+
+            if (cfg.hasOwnProperty("description") && typeof cfg.description !== "undefined" && cfg.description !== null) {
+                this.description = cfg.description;
+            }
+            if (cfg.hasOwnProperty("fileUrl") && typeof cfg.fileUrl !== "undefined" && cfg.fileUrl !== null) {
+                this.fileUrl = cfg.fileUrl;
+            }
+
+            for (i = 0; i < directProps.length; i += 1) {
+                if (cfg.hasOwnProperty(directProps[i]) && cfg[directProps[i]] !== null) {
+                    this[directProps[i]] = cfg[directProps[i]];
+                }
+            }
+
+            if (cfg.hasOwnProperty("content") && typeof cfg.content !== "undefined" && cfg.content !== null) {
```

Comment:
Not sure why we are checking both `hasOwnProperty` and the `typeof`, generally the latter is used to check to see if a property exists. Shouldn't need both.